### ISSUE TITLE
residual variances

### ISSUE THREAD

**Issue by Lakens** [yilinandrewang/pwrSEM#2]
Using default settings, SET MODEL > Proceed > SET RESIDUAL VARIANCES FOR ME - app crashes:

Warning in simulateData(input$text1, sample.nobs = 1000) :
  lavaan WARNING: some regression coefficients are unspecified and will be set to zero
Warning in simulateData(text1.t, empirical = T, standardized = T, sample.nobs = 1000) :
  lavaan WARNING: standardized=TRUE but factor loadings are >= 1.0
Warning in simulateData(text1.t, empirical = T, standardized = T, sample.nobs = 1000) :
  lavaan WARNING: standardized=TRUE but regression coefficients are >= 1.0
Warning: Error in lav_samplestats_icov: lavaan ERROR: sample covariance matrix is not positive-definite

**Comment by yilinandrewang** [comment]
Thanks for opening this issue, Daniel! I will look into the error handling for this sequence. Did you get these error messages from the log by running the app locally?

**Comment by Lakens** [comment]
Yep, ran locally. I should maybe have checked the online version but was going through the OSF materials.

**Comment by yilinandrewang** [comment]
I have updated the app and it should no longer crash on this issue. The previous version crashed because it calculated residual variances based solely on users' input; if no parameter values were entered into the parameter table in Step 3, the app could not generate a positive-definite covariance matrix from defaults, so it crashed. I added error handling such that in the case of no user input, the app assumed all values used for calculating residuals (i.e., factor loadings, regression coefficients) were 0. Both local and online versions should reflect this change.

My apologies for the late reply! I updated the app a while ago and just got around to answering here.
